boards: m5stack: cores3: Add CoreS3 SE variant

soburi opened the pull request:
Add `se` variant to support the low-cost CoreS3 SE.

Also add grove and mbus definitions and enable the peripherals.

soburi committed:
boards: m5stack: cores3: Add CoreS3 SE variant

Add `se` variant to support the low-cost CoreS3 SE.

- Add configuration files
  Add `m5stack_cores3_procpu_se(.dts|.yaml|defconfig)` files.
  Reorganize dts files to split common parts.

- Update .yaml file
  Add gpio, can, counter, entropy, pwm, and pinmux to the supported
  feature group. Remove the `ignore_tags:` section.

- Update documents
  Add and modify information about CoreS3 SE.
  Add more description about sysbuild.

Signed-off-by: TOKITA Hiroshi <tokita.hiroshi@gmail.com>

soburi committed:
boards: m5stack: cores3: Add mbus and grove connector

Adding M-Bus and Grove connector and also add configuration
for related peripherals.

Signed-off-by: TOKITA Hiroshi <tokita.hiroshi@gmail.com>

henrikbrixandersen commented:
Please split this PR into separate PRs for each of the added features (see https://docs.zephyrproject.org/latest/contribute/contributor_expectations.html#defining-smaller-prs).

soburi commented:
> Please split this PR into separate PRs for each of the added features (see https://docs.zephyrproject.org/latest/contribute/contributor_expectations.html#defining-smaller-prs).

I split and simplified this PR.
I will address the rest of the parts after #80112 is merged.

henrikbrixandersen reviewed (commented):
Is this really a board "revision" and not a board "variant"?

soburi commented:
> Is this really a board "revision" and not a board "variant"?

That's right. I fixed it.

sylvioalves commented on boards/m5stack/m5stack_cores3/m5stack_cores3_esp32s3_procpu_se.yaml:
```diff
@@ -0,0 +1,21 @@
+testing:
+  ignore_tags:
+    - bluetooth
+    - gpio
```
Any reason to let `gpio` and `pinmux` in `ignore_tags`?

soburi replied:
No reason.
It just followed the existing m5stack_cores3_procpu.yaml.

The reason is that perhaps the gpio0 and gpio1 are not enabled.
Here, I want to focus on adding a variant, so I will submit the GPIO-enabling patch soon after it is merged and address this problem.

marekmatej replied:
I think it would be much better to amend the commit by moving this part and squash it with the last commit where you do the other changes.

soburi replied:
I am done squashing commits.
And according to this https://github.com/zephyrproject-rtos/zephyr/pull/78718#discussion_r1864274910
discussion, I removed the `ignore_tags:` section.

soburi commented:
@sylvioalves @marekmatej 
Could you review it if you can take a while?

marekmatej commented on boards/m5stack/m5stack_cores3/m5stack_cores3_esp32s3_appcpu.dts:
any reason for renaming this file? IMO it does not need to follow the SoC structure as when addressing the build target. The same applies to all renamed files.

soburi replied:
The "Board Porting Guide" 
https://docs.zephyrproject.org/latest/hardware/porting/board_porting.html
says, the filename should be, 

> As described in the Write your devicetree and Write Kconfig files sections the board default configuration is created from the files <board\>.dts / <board\>_\<qualifiers\>.dts and \<board\>_defconfig / \<board\>_\<qualifiers\>_defconfig. When building for a specific board revision, the above files are used as a starting point and the following board files will be used in addition:

and here, the `<qualifiers>` is, 
  
> Board qualifiers of the form \<soc\>/\<cpucluster\>/\<variant\> are normalized so that / is replaced with _ when used for filenames, for example: soc1/foo becomes soc1_foo when used in filenames.

So, my understanding is that the original file name is no problem for it is worked but does not follow the specs.
If following the spec requires the SoC name, "esp32s3" in this case, it is needed.

I am fixing it to comply with the specifications, but I know this is not a convention in the esp32 domain, so I will revert it if that is better.

soburi replied:
I changed the file name back to the original.
Complying with the spec may be necessary in the long term, but it would be better to do this together with the other boards at that time.

marekmatej replied:
I got your point! I too think that it would be better to address the file name format for a group of boards. Thanks for clarifying your intention and bringing up the documentation bits.

marekmatej reviewed (changes requested):
The file rename part I leave up to you to decide. The yaml changes should be unified.

kartben commented on boards/m5stack/m5stack_cores3/Kconfig:
```diff
@@ -3,5 +3,6 @@
-	default 4096 if BOARD_M5STACK_CORES3_ESP32S3_PROCPU
+	default 4096 if BOARD_M5STACK_CORES3_ESP32S3_PROCPU || \
+			BOARD_M5STACK_CORES3_ESP32S3_PROCPU_SE
 	default 256 if BOARD_M5STACK_CORES3_ESP32S3_APPCPU
```
@sylvioalves did you mean to actually approve the PR then?

sylvioalves replied:
Yes, I did. It can be implemented as is for `appcpu`. As `boards.yml` only implements the `se` version for `procpu` variant, no need for the `appcpu`.
